ccl/changefeedccl: TestChangefeedFlushesSinkToReleaseMemory failed

ccl/changefeedccl.TestChangefeedFlushesSinkToReleaseMemory [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20646277?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20646277?buildTab=artifacts#/) on master @ [54a9a86f9084fe4045e054a1e8d512d818b29baa](https://github.com/cockroachdb/cockroach/commits/54a9a86f9084fe4045e054a1e8d512d818b29baa):


```
=== RUN   TestChangefeedFlushesSinkToReleaseMemory
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/a152b17b8fdc5b459db71a1944ab6ad5/logTestChangefeedFlushesSinkToReleaseMemory1477391498
    test_log_scope.go:82: use -show-logs to present logs inline
    helpers_test.go:1391: making server as secondary tenant
    changefeed_stmt.go:1783: requireNoFeedsFail: unexpected error: receiving from node 1: recv msg error: grpc: grpc: the client connection is closing [code 1/Canceled]
    changefeed_test.go:10619: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a152b17b8fdc5b459db71a1944ab6ad5/logTestChangefeedFlushesSinkToReleaseMemory1477391498
--- FAIL: TestChangefeedFlushesSinkToReleaseMemory (1.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedFlushesSinkToReleaseMemory.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55781

--- a/pkg/ccl/changefeedccl/BUILD.bazel
+++ b/pkg/ccl/changefeedccl/BUILD.bazel
@@ -340,6 +340,7 @@ go_test(
         "//pkg/util/ctxgroup",
         "//pkg/util/duration",
         "//pkg/util/encoding",
+        "//pkg/util/grpcutil",
         "//pkg/util/hlc",
         "//pkg/util/intsets",
         "//pkg/util/ioctx",

--- a/pkg/ccl/changefeedccl/helpers_test.go
+++ b/pkg/ccl/changefeedccl/helpers_test.go
@@ -54,6 +54,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util"
+	"github.com/cockroachdb/cockroach/pkg/util/grpcutil"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/json"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -941,7 +942,7 @@ func requireNoFeedsFail(t *testing.T) (fn updateKnobsFn) {
 		`context canceled`,
 	}
 	shouldIgnoreErr := func(err error) bool {
-		if err == nil || errors.Is(err, context.Canceled) {
+		if err == nil || errors.Is(err, context.Canceled) || grpcutil.IsClosedConnection(err) {
 			return true
 		}
 		for _, ignoreErr := range ignoreErrs {

